Assets/Scripts/Networking/BatchRequest.cs

```diff
@@ -161,44 +167,80 @@ public void ExecuteBatch(Action<BatchRequest> successCallback, Action<BatchReque
 		ExecuteBatch();
 	}
 
-	private void ExecuteBatch()
+	public void ExecuteBatchAsync(Action<BatchRequest> successCallback, Action<BatchRequest> failureCallback)
+	{
+		this.successCallback = successCallback;
+		this.failureCallback = failureCallback;
+		ExecuteBatch(true);
+	}
+
+	private void ExecuteBatch(bool async = false)
 	{
-		if(status == EBatchStatus.Failed)
+		m_Async = async;
```

Comment:
@HutchinsonKevin not sure about this...

There seems to be a possibility for a batch to be executed while is already in execution, causing m_ExecuteWhenReady to be set to true. Here is "remember" the async bool by assigning it to m_Async. Once the execution is ready, it will execute the "next" using m_Async.

But it does not feel right. What if I call ExecuteBatch 3 times in a row ? the second being ExecuteBatchAsync and then a third one using ExecuteBatch? Then m_Async would be false, and the data of "async" before will be lost. 
I am missing some kind of Queue here... Obviously there was already an issue because the failureCallback and succesCallback will be overriden as well . 
So this code seems to be written to be able to handle a second ExecuteBatch wil one is already running, but not a third one...